src/http/HttpRequest.ts

```diff
@@ -27,9 +31,16 @@ export class HttpRequest implements types.HttpRequest {
     constructor(init: InternalHttpRequestInit) {
         this.#init = init;
 
-        if (init.undiciRequest) {
-            this.#uReq = init.undiciRequest;
+        if (init.proxyRequest) {
+            [this.#uReq, this.query, this.params] = this.#initStreamRequest(init);
         } else {
+            [this.#uReq, this.query, this.params] = this.#initInMemoryRequest(init);
+        }
+    }
+
+    #initInMemoryRequest(init: InternalHttpRequestInit): [uRequest, URLSearchParams, HttpRequestParams] {

```

Comment:
What do you think about changing the return type from an array to a proper type? We can still use destructuring in the constructor, but this way we have a cleaner function signature and we signal to consumers more directly why these types are being bundled together.